vue/src/modules/Workspace/article/attachment/Attachments.vue

```diff
@@ -0,0 +1,289 @@
+              <b-spinner class="align-middle"/>
+              <strong>{{ $t('Loading...') }}</strong>
+            </div>
+          </template>
+          <template v-slot:cell(checkbox)="{item}">
+            <b-form-checkbox v-model="item.selected" value="1" unchecked-value="0"/>
+          </template>
+          <template v-slot:cell(name)="articleFiles">
+            <a class="attachment-name" @click="previewAttachment(articleFiles.index)">
+              {{ articleFiles.item.label || articleFiles.item.name }}
+            </a>
+          </template>
+          <template v-slot:cell(size)="{item}">
+            {{ item.size | prettyBytes }}
+          </template>
+          <template v-slot:cell(updated_at)="{item}">
+            {{ item.updated_at | toDatetime }}
+          </template>
+          <template v-slot:cell(updated_by)="{item}">
+            {{ item.updatedBy.displayName }}
+          </template>
+          <template v-slot:cell(actions)="data">
+            <b-dropdown variant="link" toggle-class="text-decoration-none p-0" no-caret right>
+              <template v-slot:button-content>
+                <i class="fas fa-ellipsis-v"/>
+              </template>
+              <b-dropdown-item @click="showEditLabelModal(data.item)">
+                <i class="far fa-edit"/>
+                {{ $t('Edit Label') }}
+              </b-dropdown-item>
+              <AttachmentDownloadButton tag="dropdown" :file="data.item"/>
+              <AttachmentDeleteButton tag="dropdown" :fileIds="[data.item.id]" :model="currentArticle"/>
+            </b-dropdown>
+          </template>
+        </b-table>
+      </b-card-body>
+    </b-card>
+    <AttachmentPreview :model="currentArticle"/>
+    <AttachmentLabel/>
+  </div>
+</template>
+
+<script>
+import {createNamespacedHelpers} from "vuex";
+import AttachmentLabel from "./AttachmentLabel";
+import AttachmentDeleteButton from "./AttachmentDeleteButton";
+import AttachmentDownloadButton from "./AttachmentDownloadButton";
+import AttachmentPreview from "./AttachmentPreview";
+
+const {mapState, mapActions} = createNamespacedHelpers('article');
+export default {
+  name: "Attachments",
+  components: {AttachmentPreview, AttachmentDownloadButton, AttachmentDeleteButton, AttachmentLabel},
+  data() {
+    return {
+      articleFileIds: [],
+      uploadedAttachmentIds: [],
+      progress: 0,
+      sortBy: null,
+      sortDesc: null
+    }
+  },
+  computed: {
+    ...mapState(['visibleColumns', 'articleFiles', 'attachConfig', 'currentArticle']),
+    fields() {
+      return [
+        {key: 'checkbox', label: ''},
+        {key: 'name', class: 'name', sortable: true},
+        ...this.visibleColumns.filter(c => c.visible),
+        {key: 'actions', class: 'text-center'}
+      ]
+    },
+    selected() {
+      return this.articleFiles.filter(a => a.selected === '1')
+    },
+    selectedIds() {
+      return this.selected.map(a => a.id)
+    },
+  },
+  watch: {
+    progress() {
```

Comment:
@skavt You don't need watcher for progress, directly assign it to zero when it hits 100 in onUploadProgress method.